src/test/java/com/microsoft/sqlserver/jdbc/unit/lobs/LobsTest.java

```diff
@@ -579,6 +583,221 @@ public void readMultipleBlobStreamsThenCloseRS() throws Exception {
         }
     }
 
+    @Test
+    @DisplayName("testBlobNotStreaming")
+    public void testBlobNotStreaming() throws SQLException, IOException {
+        String types[] = {"varbinary(max)"};
+        table = createTable(table, types, false);
+
+        int bufferSize = 10000;
+        byte[] data = new byte[bufferSize];
+        try (Connection conn = getConnection(); PreparedStatement pstmt = conn
+                .prepareStatement("INSERT INTO " + table.getEscapedTableName() + "  VALUES(?,?)")) {
+            Blob b = conn.createBlob();
+            Constants.RANDOM.nextBytes(data);
+            b.setBytes(1, data);
+            pstmt.setInt(1, 1);
+            pstmt.setBlob(2, b);
+            pstmt.executeUpdate();
+        }
+
+        String streamString = TestUtils.addOrOverrideProperty(connectionString, "delayLoadingLobs", "true");
+        String loadedString = TestUtils.addOrOverrideProperty(streamString, "delayLoadingLobs", "false");
```

Comment:
addOrOverrideProperty actually only appends the connection property at the end, so `loadedString` ends up having both delayLoadingLobs=true and delayedLoadingLobs=false in it. I tested to make sure that the test is working as intended (as in, loadedString should honor delayLoadingLobs=false), but you should be aware of this fyi.

Reply:
Yeah the function doesn't erase the contents but our driver respects the last declared connection property for duplicates.